plugins/setup.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pip
+from setuptools import setup
+from setuptools.command.develop import develop
+from setuptools.command.install import install
+
+PACKAGE_NAME = 'flytekitplugins-parent'
+SOURCES = {
+    'flytekitplugins-hive': 'hive',
+    'flytekitplugins-notebook': 'notebook',
+    'flytekitplugins-spark': 'spark',
+    'flytekitplugins-pod': 'pod',
+    'flytekitplugins-pytorch': 'pytorch',
+    'flytekitplugins-sagemaker': 'sagemaker',
+    'flytekitplugins-sqlite3': 'sqlite3',
+    'flytekitplugins-tensorflow': 'tensorflow',
+}
+
+
+def install_all_plugins(sources, develop=False):
+    """
+    Use pip to install all plugins
+    """
+    print("Installing all Flyte plugins in {} mode".format(
+        "development" if develop else "normal"))
+    wd = os.getcwd()
+    for k, v in sources.items():
+        try:
+            os.chdir(os.path.join(wd, v))
+            if develop:
+                pip.main(['install', '-e', '.'])
+            else:
+                pip.main(['install', '.'])
+        except Exception as e:
+            print("Oops, something went wrong installing", k)
+            print(e)
+        finally:
+            os.chdir(wd)
+
+
+class DevelopCmd(develop):
+    """ Add custom steps for the develop command """
+
+    def run(self):
+        install_all_plugins(SOURCES, develop=True)
+        develop.run(self)
+
+
+class InstallCmd(install):
+    """ Add custom steps for the install command """
+
+    def run(self):
+        install_all_plugins(SOURCES, develop=False)
+        install.run(self)
+
+
+setup(
+    name=PACKAGE_NAME,
+    version="0.1.0",
+    author="flyteorg",
+    author_email="admin@flyte.org",
+    description="Macrolib's description",
+    license="TBD",
+    classifiers=[
+        'Private :: Do Not Upload to pypi server',
+    ],
+    install_requires=[],
+    cmdclass={
+        'install': InstallCmd,
+        'develop': DevelopCmd,
```

Comment:
I tried running `pip install -e .` under `plugins` but that didn't seem to install any package. Am I supposed to run `python setup.py develop`?

Reply:
`python setup.py develop` doesn't work for me either. It logs everything being OK, but nothing gets installed, apart from `flytekitplugins-parent`.

Reply:
I guess we need this step? https://github.com/vaexio/vaex/blob/master/setup.py#L33

Reply:
NVM I recreated my virtualenv and it is now working.